fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/MetaDataProtoEditor.java

```diff
@@ -47,19 +47,18 @@
      * Add a new record type to the meta-data.
      *
      * <p>
-     * Adding the record type involves three steps: the message type is added to the list of file descriptor's
-     * message types, a field of the given type is added to the union, and its primary key is set. This method takes
+     * Adding the record type involves three steps: the message type is added to the file descriptor's list of message types,
+     * a field of the given type is added to the union, and its primary key is set. This method takes
      * care of the first two. Callers must also set the primary key
```

Comment:
Sorry, I should have asked this earlier, but why doesn't this take a primary key so that this method can handle everything? And why doesn't it add the record type to the list of record type messages (and set the `since_version`)?

Reply:
The reason is, we have a `DescriptorProto` here not a `Descriptor`. As mentioned before, I didn't find an easy way to convert it to a `Descriptor`.

Reply:
And, yeah it feels awkward to me too.

Reply:
Ok, the conversion from `Descriptor` from `DescriptorProto` was not necessary. I dropped the second callback. I think the API is way nicer. Thanks for the suggestion.